src/transform-sdk/cpp/examples/identity.cc

```diff
@@ -0,0 +1,23 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <redpanda/transform_sdk.h>
+
+int main() {
+    // This is an example of copying records from one location to another.
+    redpanda::on_record_written(
+      [](redpanda::write_event event, redpanda::record_writer* writer) {
+          return writer->write(event.record);
+      });
```

Comment:
well... that's neat

Reply:
Glad you like it :)

Hopefully this is a rather easy interface for folks to use. It's very similar to what Rust/Go looks like.